### PR TITLE
Added recipe for global styles with layout

### LINKED ISSUE #14893: [docs][recipes] Styling: Add global styles with a layout component
## Summary

To improve key learning workflows in the Recipes section of the docs, the "Styling" recipe should be broken up into smaller, more actionable parts. This issue is about the second item, **global CSS files with a layout component**, coming after the standard CSS file recipe requested in https://github.com/gatsbyjs/gatsby/issues/14892.

The current recipe has this text (it should be replaced with Styling recipes inline): 

> There are so many ways to add styles to your website; Gatsby supports almost every possible option, through official and community plugins.
> - Walk through adding global styles to an example site in tutorial part two
>    - More on global styles with standard CSS files
>    - More on global styles with CSS-in-JS
>    - More on global styles with CSS files and no layout component
> - Use the CSS-in-JS library Glamor
> - Use the CSS-in-JS library Styled Components
> - Use CSS Modules

The actionable version of this **layout component CSS recipe** should show how to work with a vanilla CSS file _inline_ as concisely as possible. It should include code snippets showing a short CSS file, how it can be imported into `src/components/layout.js` (like on [docs/global-css/#adding-global-styles-with-a-layout-component](https://www.gatsbyjs.org/docs/global-css/#adding-global-styles-with-a-layout-component)), and a page where that layout component would be consumed. 

Recipes now follow this format:

- Requirements or prerequisites 
- Step-by-step instructions
- Relevant links to any relevant information
    - Link to a live example
    - More on using standard global CSS files https://www.gatsbyjs.org/docs/global-css/
    - More on layout components: https://www.gatsbyjs.org/docs/layout-components/

This recipe will go into the "Styling" section of [`docs/docs/recipes.md`](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/recipes.md), replacing the existing text. There is a template in the contributing docs: https://www.gatsbyjs.org/contributing/docs-templates/#recipes

For reference, the new recipe collection is also listed in a spreadsheet: https://docs.google.com/spreadsheets/d/1yNUQi_-3TsuDtE98hCIJQvLZ6X1YHlDmC-EJwOGxiHk/edit#gid=1252007048

### PR DESCRIPTION
**Pull request by kevee**
## Description

Added a recipe for global styles using layout components.

## Related Issues

closes #14893

### CODE REVIEW

File: docs/docs/recipes.md [hunk 1]
```diff
@@ -265,6 +265,48 @@ export default ({ pageContext: { dog } }) => (
...
 
+### Using global styles in a layout component
+
+#### Prerequisites
+
+- A [Gatsby site](/docs/quick-start/) with an index page component
+
+#### Directions
+
+You can add global styles to a [shared layout component](https://www.gatsbyjs.org/tutorial/part-three/#your-first-layout-component). This component is used for things that are common throughout the site, like a header.
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
You can add global styles to a [shared layout component](/tutorial/part-three/#your-first-layout-component). This component is used for things that are common throughout the site, like a header or footer.
```
We should use relative links for the Gatsby docs wherever possible!

File: docs/docs/recipes.md [hunk 2]
```diff
@@ -265,6 +265,48 @@ export default ({ pageContext: { dog } }) => (
...
+
+Create a new directory in your site at `/src/components`. Inside, create two files: `layout.css` and `layout.js`.
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
1. If you don't already have one, create a new directory in your site at `/src/components`. 

2. Inside the components directory, create two files: `layout.css` and `layout.js`.
```

File: docs/docs/recipes.md [hunk 3]
```diff
@@ -265,6 +265,48 @@ export default ({ pageContext: { dog } }) => (
...
+
+Add the following to `layout.css`:
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
3. Add the following to `layout.css`:
```

File: docs/docs/recipes.md [hunk 4]
```diff
@@ -265,6 +265,48 @@ export default ({ pageContext: { dog } }) => (
...
+
+```css:title=/src/components/layout.css
+body {
+  background: red;
+}
+```
+
+Edit the `layout.js` file:
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
4. Edit `layout.js` to import the CSS file and output layout markup:
```

File: docs/docs/recipes.md [hunk 5]
```diff
@@ -265,6 +265,48 @@ export default ({ pageContext: { dog } }) => (
...
+
+```jsx:title=/src/components/layout.js
+import React from "react"
+import "./layout.css"
+
+export default ({ children }) => <div>{children}</div>
+```
+
+Now edit your site's homepage at `/src/pages/index.js` and use the new layout component:
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
5. Now edit your site's homepage at `/src/pages/index.js` and use the new layout component:
```

File: docs/docs/recipes.md [hunk 6]
```diff
@@ -265,6 +265,48 @@ export default ({ pageContext: { dog } }) => (
...
+
+```jsx:title=/src/pages/index.js
+import React from "react"
+import Layout from "../components/layout"
+export default () => <Layout>Hello world!</Layout>
+```
+
+#### Additional resources
+
+- [Standard Styling with Global CSS Files](https://www.gatsbyjs.org/docs/global-css/)
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
- [Standard Styling with Global CSS Files](/docs/global-css/)
```

File: docs/docs/recipes.md [hunk 7]
```diff
@@ -265,6 +265,48 @@ export default ({ pageContext: { dog } }) => (
...
+- [More about layout components](https://www.gatsbyjs.org/tutorial/part-three)
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
- [More about layout components](/tutorial/part-three)
```

File: docs/docs/recipes.md [hunk 8]
```diff
@@ -265,6 +265,48 @@ export default ({ pageContext: { dog } }) => (
...
+
+export default ({ children }) => <div>{children}</div>
+```
+
+Now edit your site's homepage at `/src/pages/index.js` and use the new layout component:
+
+```jsx:title=/src/pages/index.js
+import React from "react"
+import Layout from "../components/layout"
+export default () => <Layout>Hello world!</Layout>
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion

export default () => <Layout>Hello world!</Layout>
```
Adding a space for readability

**Comment by kevee** [comment]
Thanks, @marcysutton for the edits! I'm going to keep diving into these recipes, so I feel better after these few go arounds about formatting, etc.

**Pull request merged**